### PR TITLE
Add scanning validity checks

### PR DESCRIPTION
**Pull request by tomasz-adam-skrzypczak**
Solves #3006

### CODE REVIEW

**Comment by tomasz-adam-skrzypczak** [comment]
I've managed to accept CLA and added my github email address as an alternative but I cannot rerun workflow (and don't want to commit pointless commits). @gmlewis could you rerun workflow for me?

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:201]:
```suggestion
	SecretScanningValidityChecks *SecretScanningValidityChecks `json:"secret_scanning_validity_checks,omitempty"`
```

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:252]:
```suggestion
// SecretScanningValidityChecks represents the state of secret scanning validity checks on a repository.
```

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:254]: "Github" => "GitHub" - but if you run the commands in step 4 of CONTRIBUTING.md, then this will most likely get fixed automatically.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:261]: I'm pretty sure this can be completely removed.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @tomasz-adam-skrzypczak !
Just one small nit, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @tomasz-adam-skrzypczak !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
Whups, please run step 4 of CONTRIBUTING.md and push the results.

**Review by rufusnufus — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @rufusnufus !
Merging.

**Pull request merged**
